integration-tests/src/exchange_asset.rs

```diff
@@ -120,6 +124,86 @@ fn hydra_should_swap_assets_when_receiving_from_acala_with_sell() {
 		// We receive about 39_101 HDX (HDX is super cheap in our test)
 		let received = 39_101 * UNITS + BOB_INITIAL_NATIVE_BALANCE + 207_131_554_396;
 		assert_eq!(hydradx_runtime::Balances::free_balance(AccountId::from(BOB)), received);
+
+		let last_swapped_events = get_last_swapped_events();
+		let last_two_swapped_events = &last_swapped_events[last_swapped_events.len() - 2..];
+		let topic_id = [
+			88, 96, 17, 161, 1, 110, 221, 66, 135, 183, 36, 117, 203, 64, 121, 121, 144, 130, 127, 130, 11, 90, 144,
+			184, 162, 244, 249, 80, 95, 193, 189, 200,
+		];
+		pretty_assertions::assert_eq!(
+			last_two_swapped_events,
+			vec![
+				RuntimeEvent::AmmSupport(pallet_amm_support::Event::Swapped {
+					swapper: TempAccountForXcmAssetExchange::get(),
+					filler: Omnipool::protocol_account(),
+					filler_type: pallet_amm_support::types::Filler::Omnipool,
+					operation: pallet_amm_support::types::TradeOperation::ExactIn,
+					inputs: vec![pallet_amm_support::types::Asset::new(ACA, 50000000000000),],
+					outputs: vec![pallet_amm_support::types::Asset::new(LRNA::get(), 49180327868852)],
+					fees: vec![Fee::new(LRNA::get(), 24590163934, Omnipool::protocol_account()),],
+					operation_id: vec![
+						ExecutionType::Xcm(topic_id, 0),
+						ExecutionType::XcmExchange(1),
```

Comment:
No topic here?

Reply:
the API of AssetExchange does the full xcm instructions, so can't have the topic ID easily

		_origin: Option<&Location>,
		give: AssetsInHolding,
		want: &Assets,
		maximal: bool,
What we could maybe do is to read the topic from our stack storage, as at this point, the Xcm([u8; 32], IncrementalIdType) should be there. Not sure if it is always there, otherwise we could do use Optional type. WDYT?